source/org/zfin/publication/repository/HibernatePublicationRepository.java

```diff
@@ -843,354 +374,198 @@ public List<Figure> getFiguresByGeneAndAnatomy(Marker marker, GenericTerm anatom
-                " from expression_experiment " +
-                " where :markerZdbID = xpatex_gene_zdb_id ";
-        }
-        // nomenclature
-        if (ActiveData.isMarker(dataType)) {
-            commonPubSQL += " union " +
-                " select ra.recattrib_source_zdb_id  " +
-                " from record_attribution ra, marker_history mh " +
-                " where mh.mhist_zdb_id  = ra.recattrib_data_zdb_id " +
-                " and  :markerZdbID = mh.mhist_mrkr_zdb_id ";
+        if (journal.getAliases() == null) {
+            Set<SourceAlias> sourceAliases = new HashSet<>();
+            sourceAliases.add(journalAlias);
+            journal.setAliases(sourceAliases);
+        } else {
+            // if alias exists do not add continue...
+            if (!journal.getAliases().add(journalAlias)) {
+                return null;
+            }
         }
-        commonPubSQL += " ) as qt where recattrib_source_zdb_id like 'ZDB-PUB%'  ";
-        return commonPubSQL;
-    }
 
-    @Override
-    public List<Publication> getPubsForDisplay(String zdbID) {
+        currentSession().save(journalAlias);
 
-        List<String> publicationIDs = HibernateUtil.currentSession()
-            .createSQLQuery(getCommonPublicationSQL(zdbID))
-            .setString("markerZdbID", zdbID)
-            .list();
+        //now handle the attribution
+     /*   String updateComment;
 
-        if (CollectionUtils.isEmpty(publicationIDs)) {
-            return new ArrayList<>();
-        }
+        updateComment = "Added alias: '" + journalAlias.getAlias() + " with no attribution";
 
-        String hql = " select p from Publication p  " +
-            " where p.zdbID in (:zdbIDs) " +
-            " order by p.publicationDate desc ";
-        List<Publication> publicationLinks = HibernateUtil.currentSession()
-            .createQuery(hql)
-            .setParameterList("zdbIDs", publicationIDs)
-            .list();
 
+        InfrastructureService.insertUpdate(journal, updateComment);
+*/
+        return journalAlias;
+    }
+
+    public PaginationResult<Figure> getFiguresByGenoExp(Genotype geno) {
+        Session session = HibernateUtil.currentSession();
 
-        // remove if not pubs
-        return publicationLinks;
+        String hql = "select distinct figure from Figure figure, ExpressionResult res, ExpressionExperiment exp," +
+                "FishExperiment fishox, Genotype geno, ExpressionResultFigure xpatfig " +
+                "where geno.zdbID = :genoID AND " +
+                "      fishox.fish.genotype = geno AND " +
+                "   res.expressionExperiment = exp AND " +
+                "   xpatfig.expressionResult = res AND " +
+                "   xpatfig.figure = figure AND " +
+                "   exp.antibody is null AND " +
+                "   exp.fishExperiment = fishox  " +
+                "order by figure.orderingLabel    ";
+        Query query = session.createQuery(hql);

```

Comment:
is this using org.hibernate.query.Query? 
org.hibernate.Query is deprecated.

Reply:
updated to use the new Query interface.